### ISSUE TITLE
Configuration option to disable drag drop files completely

### ISSUE THREAD

**Issue by ChristianStadler76** [phpmyadmin/phpmyadmin#13155]
Well, the title should say everything: I'd like to have a configuration option to disable drag and drop files support completely in phpMyAdmin. When I'm importing files I now have to click the 'Choose File'-button and then navigate manually to the file in my file-system in order to circumvent the drag and drop auto-import **feature**.

**Comment by jvitasek** [comment]
+1

**Comment by WouterWidgets** [comment]
I totally agree.
I am very used to be able to drag files directly onto an `input[type=file]` browse button.

Off-topic:
Also, when importing a new database (.sql dump with a `CREATE DATABASE` statement), the new drag-and-drop functionality won't allow me to upload the file and displays a 'select database first' message.

Regards, Wouter

**Comment by jbastan** [comment]
To make this permanent, where can I find the js file? Best I can tell, first js file is loaded by PHP here: 3rdparty/phpMyAdmin/js/get_scripts.js.php
on there at line 1058 I see "_dragenter" as the function referencing "if ($(".noDragDrop").length !== 0)"

**Comment by jbastan** [comment]
I think I found my own solution to this. Since I'm using WHM/Cpanel
Path was here: usr/local/cpanel/base/3rdparty/phpMyAdmin/js
Look at file: common.js at lines 513 - 515
I commented out 3 lines:
`// $(document).on('dragenter', PMA_DROP_IMPORT._dragenter);

// $(document).on('dragover', PMA_DROP_IMPORT._dragover);

// $(document).on('dragleave', '.pma_drop_handler', PMA_DROP_IMPORT._dragleave);`

tested and all seems to be working as expected.

**Comment by eesau** [comment]
I just accidentally dropped a file to a phpMyAdmin window where it shouldn't have gone. No confirmation whether to import the file. Really lousy user interface design here.

**Comment by ibennetch** [comment]
@eesau What do you think the solution to that should be? Would a configuration directive allowing you to disable drag and drop have helped in your situation?

**Comment by ibennetch** [comment]
I would support having a configuration directive 'drag_and_drop' or 'drag_and_drop_enabled' that would allow disabling this feature.

**Comment by jbastan** [comment]
Yes, if there was a toggle in phpMyAdmin along with the other setting, it would solve my issue. I had submitted a ticket to cpanel asking for such feature in future updates, they mostly dragged their feet except for the last person who directed me here.

It's annoying because I have multiple windows open and when I dragged a file too-fast across the browser to my FTP program, it would sometimes crash the phpMyAdmin window since I can't disable/close the gray pop-over. It doesn't always auto-close. Besides, I had no idea why it was listening/expecting a file in the first place (what format, what will it do, etc). Maybe if I was on the [Import] tab, it would be more acceptable to start the listener.

Anyway, my manual fix above solved it for me. I'm happy so far till future updates replace my customization.

**Comment by eesau** [comment]
@ibennetch Yes it would. There are two reasons I want to turn it off:

1. The obvious situation would be dropping a file containing just "DROP some_important_database;" over a production server phpMyAdmin window by mistake. The database would be dropped without confirmation, which is bad. I didn't destroy an entire database, but imported some test data to a production database, which I had to delete. The Import tab is enough for my needs.

2. I have phpMyAdmin open all the time in a browser window. When it is in the background, but still visible, dragging a file over the window grays it out with the drop files here text, and this doesn't go away when the dragging continues off the window. I have to reload the page to use phpMyAdmin again. This is frustrating.

**Comment by cretace** [comment]
Has this not been addressed yet?

**Comment by jbastan** [comment]
correct. I still have to update my files when cpanel makes major changes that replaces my previous edits. Though I've been happy with my solution above (June 30, '17).
